### PR TITLE
Adds a dot to the end of say and emote messages if they don't have dots, or have exclamation or question marks

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
In hindsight I should add more things to that list
Where were you when I made a quality PR for once?
I think I saw this on Bee once like almost a month ago and it was a neat little feature that I coded it from scratch
:cl:
 * rscadd: You will now automatically add a dot at the end of whatever you say or emote if there is no dot, question mark, or exclamation mark.

### CODE REVIEW

**Comment by ShiftyRail** [comment]
I really don't see the point of this, no pun intended
Sometimes I hash out my sentences so that they can be read more clearly

File: code/__HELPERS/text.dm [hunk 1]
```diff
@@ -324,6 +324,14 @@ proc/checkhtml(var/t)
 /proc/capitalize(var/t as text)
 	return uppertext(copytext(t, 1, 2)) + copytext(t, 2)
 
+//Adds a dot at the end of the text unless there already is a dot, question mark, or exclamation point
+/proc/add_dot(var/t as text)
```
Comment (ShiftyRail) [inline code/__HELPERS/text.dm]: `as text` isn't needed.

**Comment by brndd** [comment]
Gonna need more logic to deal with cases like messages ending in "--" to indicate it being cut off. Perhaps just check if the message ends in [A-Za-z] and add a period if it does.

**Comment by boy2mantwicethefam** [comment]
I already worked on it @brndd and I'm pushing it

**Comment by boy2mantwicethefam** [comment]
@ShiftyRail That sounds like an extremely nitpicky reason but w/e

**Comment by brndd** [comment]
I'm not sure if this is a good implementation on a technical level. If the rules get any more complicated it'd probably be better to use regex.

Anyway, I'd recommend you just check if the message ends in an alphanumeric character and append the period if it does.

**Comment by nuklearcellphoneg** [comment]
Sometimes when you say something it's not the end of the sentence

**Comment by brndd** [comment]
>Sometimes when you say something it's not the end of the sentence

Stop typing like a zoomer and this is practically never the case.

**Comment by boy2mantwicethefam** [comment]
@brndd I mean, I'm not really sure what else could be added that shouldn't have a dot added, I don't think I can even code something to check if something ends in A-Z without it being all manual and basically a very long string of || if there isn't one already, there's also parantheses and stuff that can have a dot attached at the end. What do you mean by rules, which rules?
@nuklearcellphoneg You can add a , at the end of your sentence if you want.

**Comment by nuklearcellphoneg** [comment]
> like a zoomer

boy i been rping since before you was born

**Comment by SonixApache** [comment]
I mean, sentences are already
Capitalized, like this, so
Why would it even matter if you end your sentences openly on purpose?

Just use a comma, dude.

I think this'd be a nice thing to have, since there's plenty of people that NEVER fucking punctuate.

**Comment by ShiftyRail** [comment]
>I think this'd be a nice thing to have, since there's plenty of people that NEVER fucking punctuate.

And that's their right to do so

**Comment by brndd** [comment]
>And that's their right to do so

They can keep not punctuating after this PR as usual, as "is their right". The system will just fill the punctuation in for them.

**Comment by Kammerjunk** [comment]
tbh 4 consecutive sentences ending in commas (even though the complete sentence that they form when combined might grammatically warrant as little as 0 commas),
and 4 consecutive sentences ending in full stops (even though they form one complete and unbroken sentence without those full stops),
look dumber than 4 consecutive sentences with no punctuation at the end.

**Comment by bathosbathtime** [comment]
First, we should foster a player base that is capable of typing out correct sentences without the game cleaning up after them.  'Fixing' this problem like this feels an inch closer to dialog wheels.
Second, there is a difference in tone when a sentence is left open and when a full stop is used, a tone that would no longer be available under this new system.  When used right it can make a message more effective, or make a character speaking a longer sentence over a longer period of time more immersive.
Third, no, using commas exclusively to simulate this is not a good idea, before you even suggest it, as using commas to split up a long sentence, that would be mostly only split up by time would, look bad.

**Comment by boy2mantwicethefam** [comment]
@jknpj came up with a better idea, make it a preference, so I'm gonna close this and get working at some point on it

**Pull request closed without merging**